src/sulogin-shell/sulogin-shell.c

```diff
@@ -89,15 +90,28 @@ static void print_mode(const char* mode) {
 }
 
 int main(int argc, char *argv[]) {
-        static const char* const sulogin_cmdline[] = {SULOGIN, NULL};
+        /* Note: sulogin_cmdline double-NULL is intentional!
+         * (The first null might be overwritten with --force argument)
+         */
+        static const char* sulogin_cmdline[] = {SULOGIN, NULL, NULL};
         _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
         int r;
+        const char *mode;
 
         log_set_target(LOG_TARGET_AUTO);
         log_parse_environment();
         log_open();
 
-        print_mode(argc > 1 ? argv[1] : "");
+        mode = argc > 1 ? argv[1] : "";
+        print_mode(mode);
+
+        if (streq(mode, "rescue") && getenv_bool("ENABLE_FORCED_SULOGIN") > 0) {
```

Comment:
While there are some exceptions most env vars we support use the "SYSTEMD_" prefix.

Hmm, I think the ENABLE_ prefix is kinda redundant, by dropping it we lose very little.

Also, I find it quite surprising that this env var is called "FORCED" while the option is called "--force", i.e. the "d" at the end is surprising. maybe drop that too?

i.e. my suggestion would be to call this: "SYSTEMD_SULOGIN_FORCE" or so?

Reply:
why only do this for "rescue"?

Reply:
I'd appreciate clueful suggestions on in which modes it should be acceptable to use force. I just limited to rescue because better safe than sorry (and didn't expect the first revision to be accepted anyway). Also, if it should be for both rescue and emergency mode, should they be controllable separately (via separate environment variables)?

Reply:
i don't think we need to enforce anything regarding this in the sources. I mean, people can chose where they want to add the drop-in .d/ file anyway where they set the env var. i.e. they can add it to emergency.service.d/ or rescue.service.d/ or both, it's up to them really.